### PR TITLE
Update ethereum Connector

### PR DESCRIPTION
**Pull request by hugjobk**
- Add func BackfillFrom
- Add field *ethclient.Client to ethereum.Connector

### CODE REVIEW

File: chain/ethereum/ethereum.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type ETHClient interface {
 
 type Connector struct {
 	*connector.Connector
+	*ethclient.Client
```
Comment (cnkarz) [inline chain/ethereum/ethereum.go:33]: please set a variable as `Client *ethclient.Client`

File: chain/ethereum/ethereum.go [hunk 2]
```diff
@@ -208,3 +210,25 @@ func Backfill(ctx context.Context, client *ethclient.Client, addresses []common.
...
+		return Backfill(ctx, client, addresses, logs, fromBlock, fromBlock+numBlocks)
+	case fromBlock > 0 && numBlocks == 0:
+		return Backfill(ctx, client, addresses, logs, fromBlock, 0)
+	case fromBlock == 0 && numBlocks > 0:
+		toBlock, err := client.BlockNumber(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to get block number")
+			return err
+		}
+		return Backfill(ctx, client, addresses, logs, toBlock-numBlocks, toBlock)
```
Comment (cnkarz) [inline chain/ethereum/ethereum.go:230]: please replace `toBlock` with `0`

**Review by cnkarz — CHANGES_REQUESTED** [review]

File: chain/ethereum/ethereum.go (hunk 2 above)
Comment (hugjobk) [reply chain/ethereum/ethereum.go:230]: replace toBlock with 0 will cause the Backfill function to make rpc to get the latest block again

File: chain/ethereum/ethereum.go (hunk 1 above)
Comment (hugjobk) [reply chain/ethereum/ethereum.go]: done

**Review by cnkarz — APPROVED** [review]

**Pull request merged**